modules/mini-games/tightspot.lua

```diff
@@ -0,0 +1,743 @@
+local function start(args)
+    variables["level"] = {}
+    variables["surface"] = {}
+    variables["walls"] = {}
+    local level_index = args[1]
+    variables.level = config[level_index]
+    variables.diffuclty = variables.level.money[args[2]]
+    variables.loan_price = variables.level.loan_prices[args[2]]
+    variables["surface"] = game.surfaces[variables.level["surface"]]
+    if not save["tiles"][1] then
+        level_save()
+    end
+    for i, player in ipairs(game.connected_players) do
+        player_join_game(player, i - 1)
+    end
+    local force = game.players[1].force
+    force.disable_all_prototypes()
+    local recipe_list = force.recipes
+    for index, item in pairs(variables.level.recipes) do
+        recipe_list[item].enabled = true
+    end
+
+    market_setup()
+    variables.tick = game.tick
+end
+
+local function reset_table(table)
+    for i, _ in pairs(table) do
+        table[i] = nil
+    end
+end
+
+local function getSuffix(n)
+    local lastTwo, lastOne = n % 100, n % 10
+    if lastTwo > 3 and lastTwo < 21 then
+        return "th"
+    end
+    if lastOne == 1 then
+        return "st"
+    end
+    if lastOne == 2 then
+        return "nd"
+    end
+    if lastOne == 3 then
+        return "rd"
+    end
+    return "th"
+end
+
+local function Nth(n)
+    return n .. getSuffix(n)
+end
+
+local function stop()
+    game.speed = 1
+    for i, player in ipairs(game.connected_players) do
+        player.set_controller {type = defines.controllers.god}
+        player.create_character()
+        --gui
+        local Main_gui = Gui.get_left_element(player, game_gui)
+        local table = Main_gui.container["slider"].table
+        table.visible = false
+        Gui.toggle_left_element(player, game_gui, false)
+    end
+    started[1] = false
+
+    local area = variables.level.area
+    clean_up(area)
+    for i, entity in ipairs(save.entities) do
+        local name = entity[1]
+        local position = entity[2]
+        local force = entity[3]
+        local minable = entity[4]
+        local ent = variables["surface"].create_entity {name = name, position = position, force = force}
+        ent.minable = minable
+    end
+    local scores = {}
+    for name in pairs(centers) do
+        scores[#scores + 1] = {Store.get(balances, name), name}
+    end
+    local colors = {
+        ["1st"] = "[color=#FFD700]",
+        ["2nd"] = "[color=#C0C0C0]",
+        ["3rd"] = "[color=#cd7f32]"
+    }
+    table.sort(
+        scores,
+        function(a, b)
+            return a[1] > b[1]
+        end
+    )
+    for i, score in ipairs(scores) do
+        local money = score[1]
+        local player_name = score[2]
+        local place = Nth(i)
+        if colors[place] then
+            game.print(colors[place] .. place .. ": " .. player_name .. " with " .. money .. " points.[/color]")
+        else
+            game.print("[color=#808080]" .. place .. ": " .. player_name .. " with " .. money .. " points.[/color]")
```

Comment:
Might be useful to use string.format. Example `[color=%s]%s: %s with %d points[/color]`

Reply:
this will become ussless any way (cause of multi server only print in lobby the gold data) but ill change it